src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderParser.cs

```diff
@@ -30,26 +30,35 @@ private sealed class ValueCreatorCache
             private static readonly FuncInstanceMethodInfo1<ValueCreatorCache, PrimitiveDataViewType, Func<RowSet, ColumnPipe>> _getCreatorVecCoreMethodInfo
                 = FuncInstanceMethodInfo1<ValueCreatorCache, PrimitiveDataViewType, Func<RowSet, ColumnPipe>>.Create(target => target.GetCreatorVecCore<int>);
 
-            private static volatile ValueCreatorCache _instance;
-            public static ValueCreatorCache Instance
+            private static volatile ValueCreatorCache _defaultInstance;
+            public static ValueCreatorCache DefaultInstance
             {
                 get
                 {
-                    return _instance ??
-                        Interlocked.CompareExchange(ref _instance, new ValueCreatorCache(), null) ??
-                        _instance;
+                    return _defaultInstance ??
+                        Interlocked.CompareExchange(ref _defaultInstance, new ValueCreatorCache(), null) ??
+                        _defaultInstance;
                 }
             }
 
+            public static ValueCreatorCache CreateInstanceWithDoubleParserOptions(DoubleParser.Options doubleParserOptions)
+            {
+                // REVIEW: This can be more efficient if we actually store in a cache the resulting instance of this particular doubleParserOptions
+                return new ValueCreatorCache(doubleParserOptions);
+            }
```

Comment:
I think I could store this Instance in some sort of dictionary, in order to avoid creating an instance every time a TextLoader wants to use a custom doubleParserOptions, this would address the potential performance issues. I'll do this if this approach gets accepted.

Reply:
Done.

---
In reply to: [429522921](https://github.com/dotnet/machinelearning/pull/5154#discussion_r429522921) [](ancestors = 429522921)